test/e2e/functional/tests/e2e.go

```diff
@@ -0,0 +1,179 @@
+				Expect(err).ToNot(HaveOccurred())
+				Expect(len(pods.Items)).To(Equal(int(daemonset.Status.DesiredNumberScheduled)))
+
+				for _, pod := range pods.Items {
+					Expect(pod.Status.Phase).To(Equal(corev1.PodRunning))
+				}
+			})
+		})
+
+		It("should run Ingress node firewall apply rules and check the actions", func() {
+			By("get nodes IPv4 with matching labels and ping their IPs")
+			nodes, err := testclient.Client.Nodes().List(context.Background(), metav1.ListOptions{LabelSelector: "node-role.kubernetes.io/worker"})
+			Expect(err).ToNot(HaveOccurred())
+
+			err = infwutils.RunPingTest(nodes.Items)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("creating ingress node firewall rules")
+			rules := &ingressnodefwv1alpha1.IngressNodeFirewall{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "rules1",
+				},
+				Spec: ingressnodefwv1alpha1.IngressNodeFirewallSpec{
+					NodeSelector: map[string]string{
+						"node-role.kubernetes.io/worker": "",
+					},
+					Ingress: []ingressnodefwv1alpha1.IngressNodeFirewallRules{
+						{
+							SourceCIDRs: []string{"172.16.0.0/12"},
+							FirewallProtocolRules: []ingressnodefwv1alpha1.IngressNodeFirewallProtocolRule{
+								{
+									Order: 10,
+									ICMPRule: &ingressnodefwv1alpha1.IngressNodeFirewallICMPRule{
+										ICMPType: 8,
+									},
+									Protocol: ingressnodefwv1alpha1.ProtocolTypeICMP,
+									Action:   ingressnodefwv1alpha1.IngressNodeFirewallDeny,
+								},
+								{
+									Order: 20,
+									ProtocolRule: &ingressnodefwv1alpha1.IngressNodeFirewallProtoRule{
+										Ports: "800-900",
+									},
+									Protocol: ingressnodefwv1alpha1.ProtocolTypeTCP,
+									Action:   ingressnodefwv1alpha1.IngressNodeFirewallDeny,
+								},
+							},
+						},
+					},
+					Interfaces: &[]string{
+						"eth0",
+					},
+				},
+			}
+			Eventually(func() error {
+				err := testclient.Client.Create(context.Background(), rules)
+				return err
+			}, infwutils.Timeout, infwutils.Interval).Should(Succeed())
+
+			key := types.NamespacedName{
+				Name: "rules1",
+			}
+			By("checking Ingress node firewall rules resource is created")
+			Eventually(func() error {
+				err := testclient.Client.Get(context.Background(), key, rules)
+				return err
+			}, infwutils.Timeout, infwutils.Interval).Should(Succeed())
+
+			By("checking ingress node firewall nodeState resource is created")
+			nodeStateList := &ingressnodefwv1alpha1.IngressNodeFirewallNodeStateList{}
+			Eventually(func() bool {
+				err := testclient.Client.List(context.Background(), nodeStateList)
+				if err != nil {
+					return false
+				}
+				return len(nodeStateList.Items) == len(nodes.Items)
+			}, infwutils.Timeout, infwutils.Interval).Should(BeTrue())
+
+			By("checking Ingress node firewall rules deny ping packets")
+			Eventually(func() error {
+				err := infwutils.RunPingTest(nodes.Items)
```

Comment:
This wont check all the nodes that ping doesnt work. It will only check the first node in the list. RunPingTest should always return err after the first node in the list.

Reply:
`nodesIP()` will return list of all nodes IPs maybe better if I aggr err here and just return at the end ?